### PR TITLE
Demo - [Guidelines] - Public API route + logging user data + no unit test for new service

### PR DESCRIPTION
**Pull request by WilmsJochen**
### :bulb: PR Summary generated by FirstMate

**New API Route and User Data Logging Implementation**

**Changes**:
*New API Route:*
- Added `/user-data/:id` route in `exampleRouter.js` to retrieve user data.
  
*New Service Class:*
- Introduced `NewService` class in `newService.js` for handling user data retrieval.
- Implemented `getById` and `getDataFromRepo` methods with logging functionality.

*Logging Enhancements:*
- Integrated logging in `exampleService.js` to log user data when fetched.

*CI Pipeline Update:*
- Updated `.github/workflows/firstmate.yaml` to run tests after package installation.

**TLDR**: This PR adds a new API route for user data retrieval, introduces a new service class with logging, and updates the CI pipeline to include tests. Focus on the new route and service implementations.

<sub><sup>Generated by FirstMate and automatically updated on every commit.</sup></sub>

### CODE REVIEW

File: src/services/exampleService.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 export class ExampleService {
 
     async getById(id) {
         return await exampleRepo.getById(id)
     }
 
+    async getUserData(id) {
+        const user = await exampleRepo.getById(id);
+        logger.info(user.data);
+    }
```
[the remark this answers is not in the document]
Comment (WilmsJochen) [reply src/services/exampleService.js:13]: Good catch

[the remark this answers is not in the document]
Comment (WilmsJochen) [reply src/services/exampleService.js:13]: Good catch

[the remark this answers is not in the document]
Comment (WilmsJochen) [reply src/services/exampleService.js:13]: Good catch

[the remark this answers is not in the document]
Comment (WilmsJochen) [reply src/services/exampleService.js:13]: Good catch

[the remark this answers is not in the document]
Comment (WilmsJochen) [reply src/services/exampleService.js:13]: good catch

[the remark this answers is not in the document]
Comment (WilmsJochen) [reply src/services/exampleService.js:13]: good catch

[the remark this answers is not in the document]
Comment (WilmsJochen) [reply src/services/exampleService.js:13]: Good catch
